source/common/http/conn_pool_grid.cc

```diff
@@ -349,11 +349,44 @@ bool ConnectivityGrid::isPoolHttp3(const ConnectionPool::Instance& pool) {
   return &pool == pools_.begin()->get();
 }
 
-bool ConnectivityGrid::isHttp3Broken() const { return http3_status_tracker_.isHttp3Broken(); }
+absl::optional<AlternateProtocolsCache::Origin> ConnectivityGrid::originFromHost() const {
+  if (host_->address()->type() != Network::Address::Type::Ip) {
+    ENVOY_LOG_MISC(error, "Address is not an IP address");
+    ASSERT(false);
+    return absl::nullopt;
```

Comment:
I might be inclined to use a port of 0 in this case so that we can avoid having this code return an optref and instead just return a ref.